apigee-cf-service-broker/api/service_instances.js

```diff
@@ -120,6 +120,32 @@ function actionValidate(req, res, next) {
   }
 }
 
+function deriveProtocol(params){
+  //default to https if no protocol is provided
+  const proto = params.protocol ? params.protocol.toString().trim().toLowerCase() : 'https'
+  const ret = {
+    errors: []
+  }
+  if (proto === "http" || proto === "https"){
+    ret.protocol = proto
+  }
+  else if (params.protocol){
+    ret.errors.push(params.protocol.toString().split(/[\s,]+/))
```

Comment:
Do you think its best keep it as an array or just return the string the user inputted as incorrect? Because I think it might just make more sense to return the string back since it's not possible to more than one protocol whereas with the `deriveAction` made sense because you could do multiple actions in one go.